From b743ce0a1dde67ecb36c64df87fca2b0a33ed835 Mon Sep 17 00:00:00 2001
From: Krzysztof Krakowiak <krzysztof.krakowiak@gmail.com>
Date: Wed, 23 Nov 2011 13:47:11 +0100
Subject: [PATCH] replaced broken if-return logic for fixating rate and number
 of channels that caused that modules were always(after
 successfull fixation of rate) played as mono (instead of
 stereo) by correct one with appropiate warnings.

---
 ext/modplug/gstmodplug.cc |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/ext/modplug/gstmodplug.cc b/ext/modplug/gstmodplug.cc
index 6faaa0c..74b3a79 100644
--- a/ext/modplug/gstmodplug.cc
+++ b/ext/modplug/gstmodplug.cc
@@ -466,10 +466,10 @@ gst_modplug_fixate (GstPad * pad, GstCaps * caps)
   GstStructure *structure;
 
   structure = gst_caps_get_structure (caps, 0);
-  if (gst_structure_fixate_field_nearest_int (structure, "rate", 44100))
-    return;
-  if (gst_structure_fixate_field_nearest_int (structure, "channels", 2))
-    return;
+  if (!gst_structure_fixate_field_nearest_int (structure, "rate", 44100))
+    GST_WARNING_OBJECT (pad, "Failed to fixate rate to 44100");
+  if (!gst_structure_fixate_field_nearest_int (structure, "channels", 2))
+    GST_WARNING_OBJECT (pad, "Failed to fixate number of channels to stereo");
 }
 
 static gboolean
-- 
1.7.7.3

